account_index_based_currency/models/account_invoice.py

```diff
@@ -0,0 +1,130 @@
+                    inv.amount_total_signed, inv.index_based_currency_id,
+                    inv.company_id, inv.date_invoice or fields.Date.today())
+            inv.index_based_currency_amount = index_based_currency_amount
+            inv.agreement_currency_amount = (
+                inv.index_based_currency_amount / inv.agreement_currency_rate)
+
+    @api.onchange('agreement_currency_rate')
+    def onchange_agreement_currency_rate(self):
+        if self.agreement_currency_rate <= 0:
+            raise UserError(_('Invalid currency rate value'))
+
+    @api.onchange('agreement_currency_id')
+    def onchange_agreement_currency_id(self):
+        self.agreement_currency_rate = self.agreement_currency_id._convert(
+            1, self.index_based_currency_id,
+            self.company_id,
+            self.date_invoice or fields.Date.today(), round=False)
+        self.agreement_currency_amount = self.index_based_currency_id._convert(
+            self.index_based_currency_amount, self.agreement_currency_id,
+            self.company_id,
+            self.date_invoice or fields.Date.today(), round=False)
+
+    @api.model
+    def default_get(self, default_fields):
+        res = super(AccountInvoice, self).default_get(default_fields)
+        default_rate = (
+            self.env.user.company_id.currency_id._convert(
+                1, self.env.user.company_id.index_based_currency_id,
+                self.env.user.company_id,
+                self.date_invoice or fields.Date.today(), round=False))
+        res['agreement_currency_id'] = self.env.user.company_id.currency_id.id
+        res['agreement_currency_rate'] = default_rate
+        return res
+
+    agreement_currency_id = fields.Many2one(
+        'res.currency',
+        copy=True,
+        help="Currency at which the agreement is to be/was settled",
+        readonly=True, states={'draft': [('readonly', False)]},
+    )
+    index_based_currency_id = fields.Many2one(
+        'res.currency',
+        related='company_id.index_based_currency_id',
+        help="Currency used for reporting purposes",
+        copy=False)
+    index_based_currency_amount = fields.Monetary(
+        currency_field='index_based_currency_id',
+        store=True, compute='_compute_currency_amount',
+        copy=False,
+        help="Total amount in the currency of the company, negative for "
+        "credit notes.")
+    agreement_currency_amount = fields.Monetary(
+        currency_field='agreement_currency_id',
+        store=True, compute='_compute_currency_amount',
+        copy=False,
+        help="Total amount in the currency of the company, negative for "
+        "credit notes.")
+    index_based_currency_rate = fields.Float(
+        default=1,
+        store=True, compute='_compute_currency_rate',
+        help='Technical field for Index based currency rate',
+        copy=False, digits=(12, 6))
+    currency_rate = fields.Float(
```

Comment:
Is not the same field this one and the new one defined from:
https://github.com/OCA/account-invoicing/blob/0ab1d34c5c78869cad3c0859a5dab4de7d8045ff/account_invoice_change_currency/models/account_change_currency.py#L10

Reply:
Forget it
I'm seeing that this is a compute field.